### PR TITLE
[Feature][Support Struct] Support ORC Struct subfield lazy load.

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

In this pr:
1. I add a LoadType class, CompoundLoad is used for the struct subfield lazy load, which means this field's subfield will contain both lazy load field and no lazy load field.
2. I moved LazyLoad test cases out of the `apache-orc` folder, so that `run-ut.sh` can test orc lazy load directly.
3. I add a struct subfield lazy load test.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/test/formats/orc/orc_test_util/MemoryInputStream.cc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2022-present, StarRocks Inc.
```
Comment (dirtysalt) [inline be/test/formats/orc/orc_test_util/MemoryInputStream.cc:1]: There are memory input stream implementation in orc. Don't copy it.

File: be/test/formats/orc/orc_lazy_load_test.cpp [hunk 2]
```diff
@@ -0,0 +1,295 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2022-present, StarRocks Inc.
```
Comment (dirtysalt) [inline be/test/formats/orc/orc_lazy_load_test.cpp:1]: You don't have to move orc lazy load test here. Stay it in apache-orc directory.

File: be/test/formats/orc/orc_test_util/MemoryInputStream.cc (hunk 1 above)
Comment (Smith-Cruise) [reply be/test/formats/orc/orc_test_util/MemoryInputStream.cc:1]: I can't use MemoryInputStream directly, it's in `/apache-orc` test folder, and will not be included by StarRocks.

File: be/src/formats/orc/apache-orc/c++/include/orc/LoadType.h [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2022-present, StarRocks Inc.
+
+#pragma once
+
+namespace orc {
+
+enum class LoadType {
+    NO_LAZY_LOAD = 0,
```
Comment (dirtysalt) [inline be/src/formats/orc/apache-orc/c++/include/orc/LoadType.h:8]: I don't understand why there is `no_lazy_load` in it.  My intutiion is we only need `bool` to represent it(2 states). Could you make confirmation we have to use 3 states?

Comment (Smith-Cruise) [reply be/src/formats/orc/apache-orc/c++/include/orc/LoadType.h:8]: {
  a: int,
  b: {
    c: int,
    d: double
  }
}

For example, I want to set `c` as an active field, `d` as a lazy load field.
If I set `b` as an active field, `d` will not be used when we do `loadLazyLoadNext()`.
If I set `b` as a lazyload field, `c` will not be used when we do `next()`.

So we need the third state for this situation.

File: be/test/formats/orc/orc_test_util/MemoryInputStream.cc (hunk 1 above)
Comment (dirtysalt) [reply be/test/formats/orc/orc_test_util/MemoryInputStream.cc:1]: Yes, put orc lazy load test in apache-orc directory.

File: be/src/formats/orc/apache-orc/c++/include/orc/LoadType.h (hunk 3 above)
Comment (dirtysalt) [reply be/src/formats/orc/apache-orc/c++/include/orc/LoadType.h:8]: fields can be marked as lazy or not indepdently.

If c is active field, d is not. then b should be active field. For any compound type, if any subfield is acitve field, then this compound type is active.

**Review by dirtysalt — APPROVED** [review]

File: be/test/formats/orc/orc_lazy_load_test.cpp [hunk 4]
```diff
@@ -96,8 +92,107 @@ TEST(TestLazyLoad, TestNormal) {
     }
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply be/test/formats/orc/orc_lazy_load_test.cpp:92]: I think the comment is wrong 😅

[the remark this answers is not in the document]
Comment (zombee0) [reply be/test/formats/orc/orc_lazy_load_test.cpp:92]: me too

**Pull request closed without merging**